tests/test_evaluation.py

```
@@ -467,10 +481,47 @@ def test_evaluator_metrics(minimal_instance):
+
+#     # Check that the expected log message was captured in standard error
+#     assert "Empty Frame Pairs. No match found for the video frames" in err
+def test_evaluator_logging_empty_frame_pairs(caplog, minimal_instance):
+    """Test that the Evaluator logs an error when there are no matching frame pairs."""
+
+    # Create user_labels and pred_labels that will lead to empty frame pairs
+    user_labels, pred_labels = create_labels_no_match_frame_pairs(minimal_instance)
+
+    # Use caplog to capture output
+    with caplog.at_level("ERROR"):  # Set the log level to capture ERROR messages
+        with pytest.raises(Exception):
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)